doc/Migration_guide_0.4.md

```diff
@@ -0,0 +1,84 @@
+# Migrating Java service to v0.4
+
+## Update the client apps
+Update the clients to submit transactions in the new format, introduced in Exonum 0.10.
+The most notable change in the format is that the public key of the transaction author 
+must always be included in the transaction message. This allows the framework to verify 
+the message signature with no conversion to an executable transaction.
+
+If the client application is in Java, use `com.exonum.binding.common.message.TransactionMessage`;
+if it is in Java Script, use the latest version of the JS light client.
+
+The light client shall submit transactions directly to the core instead of a custom endpoint
+in the service controller. <!-- TODO: which core endpoint? -->
+
+## Update the service
+### Use the new message format
+
+EJB 0.4 accepts the transactions via a system endpoint and passes the payload to the `Service`
+implementation to convert into an executable transaction. The `TransactionConverter` now accepts
+a `RawTransaction` that includes only the transaction identifiers (service id and transaction id)
+and its payload — the serialized transaction parameters (equivalent to `Message#getBody` from v0.3).
+
+Also, consider removing the public key of the transaction author from its _payload_ because
+it is now included in a standard field of the message, and is accessible in `Transaction#execute`
+from the passed context: `TransactionContext#getFooKey`.
+
+### Update `Transaction`s
+The following methods are **removed** from `Transaction` interface:
+  - `isValid`. In v0.4 the framework verifies the cryptographic signature itself, using the
+  public key from the message. If your service performs any additional checks, 
+  move them to the constructor of your transaction or to `TransactionConverter`, whichever 
+  works best.
+  - `info`. It is no longer possible to serialize the transaction in JSON in core endpoints.
+  <!-- ^ Or shall we keep it? -->
+  - `getMessage`. Transactions are no longer required to store the corresponding 
+  `TransactionMessage`, which makes it easier to instantiate them in unit or integration tests.
+  If your transaction code needs a whole message of this or _any_ other transaction,
+  use `com.exonum.binding.blockchain.Blockchain.getTxMessages`.
+  - `hash`. You can access the SHA-256 hash of this transaction message in `Transaction#execute`
+  from the passed context: `TransactionContext#getFooHash`. As a reminder, it uniquely identifies
```

Comment:
getFooHash?

Reply:
It has not landed yet :upside_down_face:

Reply:
`getTransactionMessageHash`